components/DeleteConfirmation/index.js

```diff
@@ -83,26 +88,35 @@ export default function DeleteConfirmation({
       </StyledDeleteIcon>
 
       {showConfirmButtons && (
-        <StyledDeleteConfirmation>
-          <span>Confirm Delete</span>
-          <StyledSmallCancelButton
-            type="button"
-            onClick={() => setShowConfirmButtons(false)}
-          >
-            <Icon variant="cancel" color="var(--primaryButtonColor)" />
-            {/* Cancel */}
-          </StyledSmallCancelButton>{" "}
-          <StyledConfirmButton
-            type="button"
-            onClick={() => {
-              onDetailsPage && router.push("/");
-              onDeleteProduct(product._id);
-            }}
-          >
-            <Icon variant="delete" color="var(--primaryButtonColor)" />
-            {/* Confirm */}
-          </StyledConfirmButton>
-        </StyledDeleteConfirmation>
+        <>
+          <StyledDeleteWarningMessage>
+            {store && (
+              <>
+                <Icon variant="warning" color="var(--dangerColor)" size="25" />
+                <span>Connected products will also lose this store</span>
+              </>
+            )}
+          </StyledDeleteWarningMessage>
+          <StyledDeleteConfirmation>
+            <span>Confirm Delete</span>
+            <StyledSmallCancelButton
+              type="button"
+              onClick={() => setShowConfirmButtons(false)}
+            >
+              <Icon variant="cancel" color="var(--primaryButtonColor)" />
+            </StyledSmallCancelButton>{" "}
+            <StyledConfirmButton
+              type="button"
+              onClick={() => {
+                onDetailsPage &&
+                  (store ? router.push("/stores") : router.push("/"));
+                store ? onDeleteStore(store._id) : onDeleteProduct(product._id);
+              }}
+            >
```

Comment:
This anonymous function seems to be doing quite alot now and is getting harder to read. Maybe it is worth creating a separate function like `handleConfirmClick` and doing the logic in there?

Maybe something like. It is more code, but potentially easy to figure out what is going on.

``` js
function handleConfirmClick() {
    if (onDetailsPage) {
       const destination = store ? "/stores" : "/";
       router.push(destination);
    }

    if (store) {
        onDeleteStore(store._id);
    } else {
        onDeleteProduct(product._id);
    }
};


```